packages/slice-machine/server/src/api/screenshots/generate.ts

```diff
@@ -59,37 +59,40 @@ export async function generateScreenshot(
     (v: Models.VariationAsObject) => v.id
   );
 
-  const promises: Promise<ScreenshotUI | Error>[] = variationIds.map(
+  const promises: Promise<ScreenshotUI>[] = variationIds.map(
     (id: Models.VariationAsObject["id"]) =>
       generateForVariation(env, libraryName, slice, id)
   );
-  const results: (ScreenshotUI | Error)[] = await Promise.all(promises);
+
+  const results = await Promise.allSettled(promises);
 
   return results.reduce(
     (
       acc: ScreenshotResults,
-      screenshotOrError: ScreenshotUI | Error,
+      result: PromiseSettledResult<ScreenshotUI>,
       index: number
     ) => {
-      if (screenshotOrError instanceof Error) {
-        return {
-          screenshots: acc.screenshots,
-          failure: [
-            ...acc.failure,
-            { error: screenshotOrError, variationId: variationIds[index] },
-          ],
-        };
-      }
-
-      return {
-        screenshots: {
-          ...acc.screenshots,
-          [variationIds[index]]: screenshotOrError,
-        },
-        failure: acc.failure,
-      };
+      const key = variationIds[index];
+
+      const screenshots =
+        result.status === "fulfilled"
```

Comment:
Interesting type the PromiseSettledResult, I'm a bit sad we can't match on `GoodResult.status` instead of a string.

Reply:
Yeah, it's a bit weird